src/main/java/com/flightstats/hub/util/RequestMetric.java

```diff
@@ -0,0 +1,71 @@
+package com.flightstats.hub.util;
+
+import lombok.extern.slf4j.Slf4j;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+@Slf4j
+public class RequestMetric {
+    private final HubRequest request;
+
+    public RequestMetric(HubRequest request) {
+        this.request = request;
+    }
+
+    public RequestMetric(ContainerRequestContext requestContext) {
+        this.request = new HubRequest.RequestBuilder(requestContext).build();
+    }
+
+    public Map<String, String> getTags() {
+        Map<String, String> tags = new HashMap<>();
+        tags.put("method", request.getMethod());
+        tags.put("call", getCall());
+
+        if (!request.isInternal()) {
+            getChannelTag().ifPresent(name -> tags.put("channel", name));
+        }
+
+        return tags;
+    }
+
+    public Optional<String> getMetricName() {
+        if (!request.getEndpoint().isPresent()) {
+            log.trace("no endpoint, path: {}", request.getPath());
+            return Optional.empty();
+        } else if (request.isShutdown()) {
+            log.info("call to shutdown, ignoring statsd time");
+            return Optional.empty();
+        }
+
+        String requestScope = request.isInternal() ? "internal" : "api";
+        String channelDifferentiator = request.isChannelRelated() ? "channel" : "nonchannel";
+        return Optional.of(String.format("request.%s.%s", requestScope, channelDifferentiator));
+    }
+
+    public boolean shouldReport(Set<String> metricsToIgnore, Function<String, Boolean> testChannelCheck) {
```

Comment:
It feels a little weird to have a metric know whether or not it should report itself. I am really only saying this because the signature of this function feels funny to me.

Reply:
It made it easier to test, but then I refactored the calling method to make it more testable. Totally makes sense to move it back to where it belongs now that testing's not a pain.